src/prometheus/definition.go

```diff
@@ -540,6 +558,82 @@ func FromSummary(key string) definition.FetchFunc {
 	}
 }
 
+// FromValueWithLabelsFilter creates a FetchFunc that fetches values from prometheus metrics values given specific
+// labels filter.
+func FromValueWithLabelsFilter(metricName string, nameOverride string, labelsFilter ...LabelsFilter) definition.FetchFunc {
+	return func(groupLabel, entityID string, groups definition.RawGroups) (definition.FetchedValue, error) {
+		value, err := definition.FromRaw(metricName)(groupLabel, entityID, groups)
+		if err != nil {
+			return nil, err
+		}
+
+		switch m := value.(type) {
+		case Metric:
+			return m.Value, nil
+		case []Metric:
+			return fetchedValuesFromRawMetricsWithoutSuffix(metricName, nameOverride, m, labelsFilter...)
+		}
+		return nil, fmt.Errorf(
+			"incompatible metric type for %s. Expected: Metric or []Metric. Got: %T",
+			metricName,
+			value,
+		)
+	}
+}
+
+func fetchedValuesFromRawMetricsWithoutSuffix(
```

Comment:
perhaps we could call this `fetchedValuesFromRawMetrics` and change the other something like `fetchedValuesFromRawMetricsAddingSuffix` to make things more clear WDYT

Reply:
also perhaps we could extract some of the shared logic into functions that both could share

Reply:
I agree, updated it here 97e8b42149397faa07c96010501130a1bf0c1906.